src/server/active_tran_server.cpp

```diff
@@ -189,7 +211,23 @@ active_tran_server::connection_handler::on_connecting ()
 
   m_prior_sender_sink_hook_func = std::bind (&active_tran_server::connection_handler::prior_sender_sink_hook, this,
 				  std::placeholders::_1);
-  log_Gl.m_prior_sender.add_sink (m_prior_sender_sink_hook_func);
+
+  auto unsent_lsa = log_Gl.m_prior_sender.add_sink (m_prior_sender_sink_hook_func);
+
+  std::string hostname = "N/A";
+  int32_t port = -1;
+
+  if (m_ts.get_main_connection_info (hostname, port) == false)
+    {
+      // there is no main connection yet. it's booting up.
+      // TODO: We should make two paths to set a connection_handler to CONNECTED:
+      //    - While booting: a target PS is determined after communicating with multiple PSes and
+      //      send the catch-up request based on it.
+      //    - While running: the unsent_lsa from prior_sender is the LSA of the first log records to send.
+      unsent_lsa.set_null ();
```

Comment:
Others besides the leader PS can do the "cold-start". It'd be better to send just LSA and let the PS decide if it's cold-start or not.